src/chapters/az-cli/walkthrough_az-cli.rst

```diff
@@ -0,0 +1,537 @@
+
+Exploration
+-----------
+
+Virtual Machines are naturally more complex to interact with than a simple resource group. However, now that we understand the pattern of the ``az CLI`` that complexity can be managed using the help command to methodically work our way through its Sub-Groups and Commands.
+
+Once again let's begin by assessing what is available to us:
+
+.. sourcecode:: powershell
+
+    > az vm -h
+
+Planning
+--------
+
+Creating a VM will naturally require many Arguments to customize it. Recall in the web portal how there were several menus we had to work through to provision it. In addition to all of those options the ``az CLI`` exposes additional configuration Arguments for more granular control. 
+
+Let's see what Arguments are associated with creating a VM:
+
+.. sourcecode:: powershell
+
+    > az vm create -h
+
+From this long list of arguments we will need to provide values for the following:
+
+- ``-n``: the name of the VM
+- ``-l``: the location [default configured]
+- ``-g``: the resource group name [default configured]
+- ``--size``: the size of the VM
+- ``--image``: the URN of the image used to create the VM
+- ``--admin-username``: the username of the root account for the VM
+- ``--assign-identity``: to assign an identity to the VM for granting access to the KeyVault secrets
+
+Listing Images
+^^^^^^^^^^^^^^
+
+In order to define the image for the VM we have to find its URN. In the ``vm create`` help output we saw a note that guided is in discovering these URN values. Let's list the available images using the ``vm`` Sub-Group ``image`` and its associated ``list`` Command:
+
+.. sourcecode:: powershell
+
+    > az vm image list
+
+Many different images are provided in the JSON object list output. But all we care for is the URN values. We could manually scroll through all of them to find the URN of the Ubuntu image. Or we can make use of the global ``--query`` Argument to output only the data we need!
+
+The `JMESPath query<https://jmespath.org/>`_ value we will use is ``"[].urn"`` which means take the output list ``[]`` and instead of the complete image objects only output the value for each of their the ``urn`` properties. The result is a list of just URN values which is much easier to work with!
+
+.. sourcecode:: powershell
+
+    > az vm image list --query "[].urn"
+
+From here we can see the URN we need for the Ubuntu image is ``"Canonical:UbuntuServer:18.04-LTS:latest"``. Let's assign that value to a variable so we don't have to clutter our clipboard:
+
+.. sourcecode:: powershell
+    :caption: on Windows/PowerShell
+
+    > $ImageURN="Canonical:UbuntuServer:18.04-LTS:latest"
+
+.. sourcecode:: bash
+    :caption: on Linux/BASH
+
+    $ image_urn="Canonical:UbuntuServer:18.04-LTS:latest"
+
+Now we can reference the URN by its variable name ``$ImageURN`` or ``image_urn`` depending on your chosen shell.
```

Comment:
suggested change from "...name ``$ImageURN`` or ``image_urn`` depending..." to "...name ``$ImageURN`` (powershell) or ``image_urn`` (bash) depending..." make it explicit just to reiterate the point.